lib/Services/AppSettingsService.php

```diff
@@ -0,0 +1,196 @@
+    }
+
+    /**
+     * Get the subject of the images to use
+     *
+     * @return string
+     */
+    public function getImageSubject(): string {
+        return $this->config->getAppValue($this->appName, self::IMAGE_SUBJECT, 'any');
+    }
+
+    /**
+     * Set the subject of the images to use
+     *
+     * @param string $subject
+     */
+    public function setImageSubject(string $subject) {
+        $this->config->setAppValue($this->appName, self::IMAGE_SUBJECT, $subject);
+    }
+
+    /**
+     * Get the amount of images to be prefetch
+     *
+     * @return int
+     */
+    public function getImageAmount(): int {
+        return $this->config->getAppValue($this->appName, self::IMAGE_AMOUNT, 32);
+    }
+
+    /**
+     * Set the amount of images to be prefetch
+     *
+     * @param int $amount
+     */
+    public function setImageAmount(int $amount) {
+        $this->config->setAppValue($this->appName, self::IMAGE_AMOUNT, $amount);
+    }
+
+    /**
+     * Get the service to provide images
+     *
+     * @return string
+     */
+    public function getImageProvider(): string {
+        return $this->config->getAppValue($this->appName, self::IMAGE_PROVIDER, 'unsplash');
+    }
+
+    /**
+     * Set the service to provide images
+     *
+     * @param string $provider
+     */
+    public function setImageProvider(string $provider) {
+        $this->config->setAppValue($this->appName, self::IMAGE_PROVIDER, $provider);
+    }
+
+    /**
+     * If the image should be persistent throughout the whole session
+     *
+     * @return bool
+     */
+    public function imagePersistenceEnabled(): bool {
+        $styleHeader = $this->config->getAppValue($this->appName, self::IMAGE_PERSISTENCE, 1);
+
+        return $styleHeader == 1;
+    }
+
+    /**
+     * Set if the image should be persistent throughout the whole session
+     *
+     * @param bool $imagePersistence
+     *
+     * @return void
+     */
+    public function setImagePersistenceEnabled(bool $imagePersistence = true) {
+        $this->config->setAppValue($this->appName, self::IMAGE_PERSISTENCE, $imagePersistence ? 1:0);
+    }
+
+    /**
+     * If the users are allowed to set their own subject
+     *
+     * @return bool
+     */
+    public function allowUserSubjects(): bool {
+        return $this->config->getAppValue($this->appName, self::USER_SUBJECT, 0) == 1;
+    }
+
+    /**
+     * Set if the users are allowed to set their own subject
+     *
+     * @param bool $allowUserSubject
+     */
+    public function setAllowUserSubjects(bool $allowUserSubject = false) {
+        $this->config->setAppValue($this->appName, self::USER_SUBJECT, $allowUserSubject);
+    }
+
+    /**
+     * Get the api key to use for the unsplash api
+     *
+     * @return string
+     */
+    public function getApiKey(): string {
+        return $this->config->getAppValue($this->appName, self::API_KEY, 'ed3b10c059c5da43fe16c34ce5467b6c472abc10d21c498a6159df9fb556930a');
```

Comment:
There is an api-key, it shouldn't be there, right? ;)

Reply:
Actually it is there on purpose. The api key is required, but if the user does not specify an api key a default one is used.
But i think that maybe i should hide it in the settings.